homeassistant/components/sensor/ebusd.py

```diff
@@ -0,0 +1,251 @@
+        ['hwcTimer.Friday', None, 'mdi:timer', 1],
+    'HWTimerSaturday':
+        ['hwcTimer.Saturday', None, 'mdi:timer', 1],
+    'HWTimerSunday':
+        ['hwcTimer.Sunday', None, 'mdi:timer', 1],
+    'WaterPressure':
+        ['WaterPressure', 'bar', 'mdi:water-pump', 0],
+    'Zone1RoomZoneMapping':
+        ['z1RoomZoneMapping', None, 'mdi:label', 0],
+    'Zone1NightTemperature':
+        ['z1NightTemp', '°C', 'mdi:weather-night', 0],
+    'Zone1DayTemperature':
+        ['z1DayTemp', '°C', 'mdi:weather-sunny', 0],
+    'Zone1HolidayTemperature':
+        ['z1HolidayTemp', '°C', 'mdi:thermometer', 0],
+    'Zone1RoomTemperature':
+        ['z1RoomTemp', '°C', 'mdi:thermometer', 0],
+    'Zone1ActualRoomTemperatureDesired':
+        ['z1ActualRoomTempDesired', '°C', 'mdi:thermometer', 0],
+    'Zone1TimerMonday':
+        ['z1Timer.Monday', None, 'mdi:timer', 1],
+    'Zone1TimerTuesday':
+        ['z1Timer.Tuesday', None, 'mdi:timer', 1],
+    'Zone1TimerWednesday':
+        ['z1Timer.Wednesday', None, 'mdi:timer', 1],
+    'Zone1TimerThursday':
+        ['z1Timer.Thursday', None, 'mdi:timer', 1],
+    'Zone1TimerFriday':
+        ['z1Timer.Friday', None, 'mdi:timer', 1],
+    'Zone1TimerSaturday':
+        ['z1Timer.Saturday', None, 'mdi:timer', 1],
+    'Zone1TimerSunday':
+        ['z1Timer.Sunday', None, 'mdi:timer', 1],
+    'Zone1OperativeMode':
+        ['z1OpMode', None, 'mdi:math-compass', 3],
+    'ContinuosHeating':
+        ['ContinuosHeating', '°C', 'mdi:weather-snowy', 0],
+    'PowerEnergyConsumptionLastMonth':
+        ['PrEnergySumHcLastMonth', 'kWh', 'mdi:flash', 0],
+    'PowerEnergyConsumptionThisMonth':
+        ['PrEnergySumHcThisMonth', 'kWh', 'mdi:flash', 0]
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CIRCUIT): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ebusd..."""
+    name = config.get(CONF_NAME)
+    circuit = config.get(CONF_CIRCUIT)
+    server_address = (config.get(CONF_HOST), config.get(CONF_PORT))
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        data = EbusdData(server_address, circuit)
+
+        sock.settimeout(5)
+        sock.connect(server_address)
+        sock.close()
+
+        dev = []
+        for variable in config[CONF_MONITORED_CONDITIONS]:
+            dev.append(Ebusd(data, variable, name))
+
+        add_devices(dev)
+        hass.services.register('sensor', 'ebusd_write', data.write)
+    except socket.timeout:
+        _LOGGER.error("socket timeout error")
+        return
+    except socket.error:
+        _LOGGER.error("socket error")
+        return
+
+
+def timer_format(string):
```

Comment:
We should prefer storing times in isoformat.

Reply:
what do you mean with ISO format? Is it only HH:mm not a datetime